<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN">
<html>
<head>
<meta name="copyright" content="2003,  Bradley Kjell" />
<meta name="author" content="Bradley Kjell kjell@ieee.org " />
<meta name="translation" content="Copyright 2004 Deutsche &Uuml;bersetzung Heinrich Gailer heinrich@gailer-net.de" />
<meta name="robots" content="index,follow" />
<title>Output Beispiele</title>
<link rel="stylesheet" type="text/css" href="../CAIstyle.css" />
</head>

<body>
<!-- generated by CSSmaker.java, version of 10/12/2004-->
<!-- from source javaChap81.frames on Wed Oct 24 11:17:05 CEST 2007 -->

<!-- ANSWER AREA -->



<div class="answer">
<h3>Eine gute Antwort w&auml;re:</h3>

<pre class="code">
main() ruft methodeA
<span style="color:blue;font-weight:bold">methodeA ruft parseInt()</span>
<span style="color:blue;font-weight:bold">parseInt() l&ouml;st eine NumberFormatException aus.</span>
<span style="color:blue;font-weight:bold">methodeA f&auml;ngt die Ausnahme ab und gibt "Falsche Eingabedaten!!" aus.</span>
<span style="color:blue;font-weight:bold">Die return-Anweisung wird ausgef&uuml;hrt.</span>
Kontrolle kehrt zu main() zur&uuml;ck.
</pre>

</div>

<!-- LESSON AREA -->
<div class="lesson">
<h1>Output Beispiele</h1>

<p>
Hier ist der Output des Programms, wenn der Anwender "0" eingibt.
Das Stack-Protokoll ist ein "Schnappschuss" der Situation zum Zeitpunkt der
Ausnahme:
</p>

<pre class="code">
Geben Sie den Divisor ein:
0
Division durch Null!!
java.lang.ArithmeticException: / by zero
        at SchwarzenPeterWeiterrreichen.methodeB(SchwarzenPeterWeiterrreichen.java:9)
        at SchwarzenPeterWeiterrreichen.methodeA(SchwarzenPeterWeiterrreichen.java:32)
        at SchwarzenPeterWeiterrreichen.main(SchwarzenPeterWeiterrreichen.java:51)
weiter mit main...!
</pre>


<p>
Hier ist der Output des Programms, wenn der Anwender "Unsinn" eingibt.
</p>

<pre class="code">
Geben Sie den Divisor ein:
Unsinn
Falsche Eingabedaten!!
java.lang.NumberFormatException: rats
        at java.lang.Integer.parseInt(Integer.java:409)
        at java.lang.Integer.parseInt(Integer.java:458)
        at SchwarzenPeterWeiterrreichen.methodeA(SchwarzenPeterWeiterrreichen.java:21)
        at SchwarzenPeterWeiterrreichen.main(SchwarzenPeterWeiterrreichen.java:51)
weiter mit main...!
</pre>


<p>
Wenn Sie das Programm in einer Datei gespeichert haben, &auml;ndern Sie es,
so dass beide Typen von Ausnahmen von <code>main()</code>
aufgefangen und behandelt werden.
Dann &auml;ndern Sie es, so dass jede Methode ihre eigenen Ausnahmen auff&auml;ngt.
Es gibt mehrere andere Varianten, an die Sie denken k&ouml;nnten. Probieren Sie sie aus.
</p>


</div>

<!-- QUESTION AREA -->
<div class="question">
<h3> FRAGE 10:</h3>
<a href="ch81_11.html">
<img src="../nextBut.gif" hspace="5" align="right" border="0"/></a>
<p>
Zeigt das Stack-Protokoll die Situation:
</p>

<ul>
<li>zum Zeitpunkt an dem die Ausnahme auftritt oder</li>
<li>zum Zeitpunkt an dem die Ausnahme aufgefangen wird.</li>
</ul>


</div>
<div>
<a title="Zurück zum Inhaltsverzeichnis gehen." href="../../java-toc.html#81">Inhaltsverzeichnis</a>
</div>

</body>
</html>
